pyphare/pyphare/pharesee/hierarchy.py

```diff
@@ -262,6 +486,99 @@ def __call__(self, qty, **kwargs):
+
+        if len(self.patch_datas.keys()) == 1:
+            names = ['value']
+        elif len(self.patch_datas.keys()) == 3:
+            names = ['x', 'y', 'z']
+        elif len(self.patch_datas.keys()) == 6:
+            names = ['xx', 'xy', 'xz', 'yy', 'yz', 'zz']
+        elif len(self.patch_datas.keys()) == 9:
+            names = ['xx', 'xy', 'xz', 'yx', 'yy', 'yz', 'zx', 'zy', 'zz']
+
+        new_pd = {}
+
+        for (name, name_self, name_operand) in zip(names, self.patch_datas.keys(), operand.patch_datas.keys()):
+            pd_self = self.patch_datas[name_self]
+            pd_operand = operand.patch_datas[name_operand]
+
+            if pd_self.centerings == pd_operand.centerings:
+                dset = np.asarray(pd_self.dataset)
+                # pd = FieldData(self.layout, name, dset, centering=pd_self.centerings)
+                # new_pd[name] = pd
+                new_pd[name] = pd_self
+
+            else:
+                pd_shape = np.empty(self.box.ndim, dtype=int)
+                pd_centerings = [None]*self.box.ndim
+
+                for i, (cent_self, cent_operand) in enumerate(zip(pd_self.centerings, pd_operand.centerings)):
+                    if cent_self == cent_operand:
+                        pd_shape[i] = pd_self.dataset.shape[i]
+                        pd_centerings[i] = cent_self
+                    else:
+                        if cent_self == "primal":
+                            pd_shape[i] = pd_self.dataset.shape[i]
+                            pd_centerings[i] = cent_self
+                        else:
+                            pd_shape[i] = pd_operand.dataset.shape[i]
+                            pd_centerings[i] = cent_operand
+
+                pd = pd_self.to_given_centering(pd_shape, pd_centerings)
+                new_pd[name] = pd
+
+        return Patch(new_pd, self.id)
+
+
+    def __mul__(self, operand):
+        """
+        implementation of the product of 2 patches
+        caveats : yet only works for patches associated to vectors, that is w. 3 names/dset
+
+        """
+
+        assert(len(self.patch_datas.keys()) == len(operand.patch_datas.keys()))
+
+        if len(self.patch_datas.keys()) == 1:
+            names = ['value']
+
+        if len(self.patch_datas.keys()) == 3:
+            names = ['x', 'y', 'z']
+
+        if len(self.patch_datas.keys()) == 6:
+            names = ['xx', 'xy', 'xz', 'yy', 'yz', 'zz']
+
+        if len(self.patch_datas.keys()) == 9:
+            names = ['xx', 'xy', 'xz', 'yx', 'yy', 'yz', 'zx', 'zy', 'zz']
+
+        left_op = self % operand
+        right_op = operand % self
+
+        new_pd = {}
+
+        for (name, name_left, name_right) in zip(names, left_op.patch_datas.keys(), right_op.patch_datas.keys()):
+            pd_left = left_op.patch_datas[name_left]
+            pd_right = right_op.patch_datas[name_right]
+
+            dset = np.asarray(pd_left.dataset)*np.asarray(pd_right.dataset)
```

Comment:
This looks like the product component-wise. I think it's misleading as this is neither the dot/cross/outer product